### PR TITLE
docs: pivotal tracker feature (diagram updated)

### PR DESCRIPTION
**Pull request by weedySeaDragon**
Updated the diagram for "How we use pivotal tracker"

**_Note: PR #138 replaces this one. (The commits here had gotten messy.)_**

Changes proposed in this pull request:
1.  changed the exit step in "Start" to _all TESTS must pass_
2. added state 'Points assigned'
2. added the 'source file' for the digram: `pivotal_tracker_feature.rb` (has the aasm state machine code in there)
4. added the configuration file used to generate the diagram (`yml` file)

Ready for review:
@patmbolger @thesuss

### CODE REVIEW

**Review by thesuss — APPROVED** [review]

**Comment by thesuss** [comment]
@weedySeaDragon Also here I think you need to start from a clean develop and cherry pick the relevant commits, here are things that should not be in this PR (i.e. the poltergeist stuff). 

If that does not help I think your develop has been compromised...

**Review by thesuss — CHANGES_REQUESTED** [review]
I take back my approval ..

File: features/support/hooks.rb [hunk 1]
```diff
@@ -6,3 +6,13 @@
   Capybara.reset_sessions!
   Capybara.current_driver = :rack_test
 end
+
```
Comment (thesuss) [inline features/support/hooks.rb:9]: @weedySeaDragon  it is here too

**Comment by patmbolger** [comment]
I am a bit confused ..... I had thought that the "ready for scrum review" state is transitioned to by the developer pressing the "Finished" button.  That is, the "ready for scrum review" state is indicated by the presence of the "Deliver" button.   The developer would do this before the sprint review meeting.

Then, during the sprint review, we decide whether, in fact, the story has indeed been delivered.  If so, we press the "Deliver" button during the meeting, and the story is now ready for client review.

That is not what the diagram shows(?).  My understanding of the STD we discussed could be wrong, of course.

**Comment by weedySeaDragon** [comment]
@patmbolger and @thesuss : I think what Pat says makes sense.  But it's not what I have in my notes.  It's quite possible that my notes are wrong.

Regardless of my notes, we should agree on what _we_ want to happen.

**When should the developer press _Finished_ in PivotalTracker?**
1. when the github merge is done?
2. when the developer thinks everything is OK, even if the merge hasn't been done yet? (because someone else has to do the merge, and so there might be a delay due to timezones and availability)
3. only after it's been discussed in the scrum and everyone is ok with it?

My vote is for **2.**  If, during the scrum we decide that there's more work to be done, we can change the state back to started.

_(and after we figure this out, I'll rebase and submit a new PR with the commits cleaned up)_

@weedySeaDragon - my vote is #2 as well.  The developer signals that he/she thinks the story is ready for review by clicking "Finish", and, of course, usually also creating a PR.  (<< Pat's comment)

**Comment by patmbolger** [comment]
BTW, I think this workflow also works for bugs.   The only difference in treatment is - as we had discussed - at the sprint review meeting we review all "Finished" bugs and follow this logic:

1) If not finished, set state back to "Start"
2) Otherwise, click "Deliver" button
3) If this bug is visible to the client, we are done (further state change depends on client accept/reject
4) If this bug **not** visible to the client, click "Accept" and close out the story

_(steps 3 and 4 are additional for "bug" stories, but not for feature stories)_

(and, I *think* this is what we had decided - wow, we can't get that documentation done fast enough!)

**Comment by weedySeaDragon** [comment]
Is the diagram in #138 close, @patmbolger ?  What do you think, @thesuss ?

That is the PR that will replace this one.  I copied the comments from there into there so they'll be easier for us to refer to.

**Pull request closed without merging**